tools/util/marshaller.go

```diff
@@ -0,0 +1,107 @@
+package util
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func MarshallObject(obj interface{}, writer io.Writer) error {
+	jsonBytes, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	var r unstructured.Unstructured
+	if err := json.Unmarshal(jsonBytes, &r.Object); err != nil {
+		return err
+	}
+
+	// remove status and metadata.creationTimestamp
+	unstructured.RemoveNestedField(r.Object, "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "spec", "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "status")
+
+	/*	// remove dataSource from PVCs if empty
+		templates, exists, err := unstructured.NestedSlice(r.Object, "spec", "dataVolumeTemplates")
+		if exists {
+			for _, tmpl := range templates {
+				template := tmpl.(map[string]interface{})
+				_, exists, err = unstructured.NestedString(template, "spec", "pvc", "dataSource")
+				if !exists {
+					unstructured.RemoveNestedField(template, "spec", "pvc", "dataSource")
+				}
+			}
+			unstructured.SetNestedSlice(r.Object, templates, "spec", "dataVolumeTemplates")
+		}*/
+	objects, exists, err := unstructured.NestedSlice(r.Object, "objects")
+	if exists {
+		for _, obj := range objects {
+			object := obj.(map[string]interface{})
+			kind, exists, _ := unstructured.NestedString(object, "kind")
+			if exists && kind == "PersistentVolumeClaim" {
+				_, exists, err = unstructured.NestedString(object, "spec", "dataSource")
+				if !exists {
+					unstructured.RemoveNestedField(object, "spec", "dataSource")
+				}
+			}
+		}
+		unstructured.SetNestedSlice(r.Object, objects, "objects")
+	}
+
+	deployments, exists, err := unstructured.NestedSlice(r.Object, "spec", "install", "spec", "deployments")
+	if exists {
+		for _, obj := range deployments {
+			deployment := obj.(map[string]interface{})
+			unstructured.RemoveNestedField(deployment, "metadata", "creationTimestamp")
+			unstructured.RemoveNestedField(deployment, "spec", "template", "metadata", "creationTimestamp")
+			unstructured.RemoveNestedField(deployment, "status")
+		}
+		unstructured.SetNestedSlice(r.Object, deployments, "spec", "install", "spec", "deployments")
+	}
+
+	// remove "managed by operator" label...
+	labels, exists, err := unstructured.NestedMap(r.Object, "metadata", "labels")
+	if exists {
+		//delete(labels, v1.ManagedByLabel)
```

Comment:
do we need to keep this commented line? If so, we should add a comment saying why

Reply:
Removed the line. as it's not required.